app/models/atheneum/command/upsert_exam.rb

```diff
@@ -22,4 +23,10 @@ def self.parse_json(exam_json)
     ['start_time', 'end_time'].each { |param| exam[param] = exam[param].to_time }
     exam
   end
+
+  def self.validate_repeated_exam(id, guide_id)
+    if Exam.find_by(classroom_id: id).blank? && Exam.find_by(guide_id: guide_id)
+      Exam.find_by(guide_id: guide_id).destroy!
+    end
```

Comment:
This method name is wrong: it is not just validating, but also removing the exam. 
Perhaps we could rename it to something like `remove_previous_versions`. 

An exam is a previous version of another when it is older, but points to the same guide. And we could just remove all the previous versions, which is simpler but also safer: 

```ruby
 if !Exam.exist(classroom_id: id)
     Exam.destroy_all!(guide_id: guide_id)
  end
```